diff --git a/Cargo.toml b/Cargo.toml
index 0946d48cf..d83ac2e30 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,6 +31,7 @@ members = [
     "kernel/core/comps/block",
     "kernel/core/comps/cmdline",
     "kernel/core/comps/console",
+    "kernel/core/comps/device-mapper",
     "kernel/core/comps/framebuffer",
     "kernel/core/comps/i8042",
     "kernel/core/comps/input",
@@ -81,6 +82,7 @@ default-members = [
     "kernel/core/comps/block",
     "kernel/core/comps/cmdline",
     "kernel/core/comps/console",
+    "kernel/core/comps/device-mapper",
     "kernel/core/comps/framebuffer",
     "kernel/core/comps/i8042",
     "kernel/core/comps/input",
@@ -160,6 +162,7 @@ aster-core = { path = "kernel/core", default-features = false }
 aster-block = { path = "kernel/core/comps/block" }
 aster-cmdline = { path = "kernel/core/comps/cmdline" }
 aster-console = { path = "kernel/core/comps/console" }
+aster-device-mapper = { path = "kernel/core/comps/device-mapper" }
 aster-framebuffer = { path = "kernel/core/comps/framebuffer" }
 aster-i8042 = { path = "kernel/core/comps/i8042" }
 aster-input = { path = "kernel/core/comps/input" }
