diff --git a/distro/etc_nixos/configuration.nix b/distro/etc_nixos/configuration.nix
index 5b1367e8d..737e63074 100644
--- a/distro/etc_nixos/configuration.nix
+++ b/distro/etc_nixos/configuration.nix
@@ -24,7 +24,15 @@
 
   # List packages installed in system profile.
   # You can use https://search.nixos.org/ to find more packages (and options).
-  environment.systemPackages = with pkgs; [ hello-asterinas ];
+  # 安装第一版 Linear Device Mapper 验收所需的标准用户态工具。
+  environment.systemPackages = with pkgs; [
+    aster-dm-disk-locator
+    e2fsprogs
+    hello-asterinas
+    lvm2
+    strace
+    util-linux
+  ];
 
   system.nixos.distroName = "Asterinas NixOS";
 
