diff --git a/distro/etc_nixos/modules/systemd.nix b/distro/etc_nixos/modules/systemd.nix
index 9e983bcb8..7f78c92c4 100644
--- a/distro/etc_nixos/modules/systemd.nix
+++ b/distro/etc_nixos/modules/systemd.nix
@@ -18,22 +18,13 @@
   services.udev.enable = false;
 
   services.getty.autologinUser = "root";
+  systemd.services."serial-getty@hvc0".enable = false;
   users.users.root = {
     shell = "${pkgs.bash}/bin/bash";
     hashedPassword = null;
   };
 
-  systemd.targets.getty.wants =
-    # tty1: provide text login ONLY when X server is disabled.
-    # Other VTs: always provide text logins
-    (lib.optional (!config.services.xserver.enable) "autovt@tty1.service") ++ [
-      "autovt@hvc0.service"
-      "autovt@tty2.service"
-      "autovt@tty3.service"
-      "autovt@tty4.service"
-      "autovt@tty5.service"
-      "autovt@tty6.service"
-    ];
+  systemd.targets.getty.wants = lib.mkForce [ "autovt@hvc0.service" ];
 
   systemd.extraConfig = ''
     LogLevel=crit
