diff --git a/kernel/comps/device-mapper/src/target/mod.rs b/kernel/comps/device-mapper/src/target/mod.rs
new file mode 100644
index 000000000..5c8b40cf3
--- /dev/null
+++ b/kernel/comps/device-mapper/src/target/mod.rs
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::ops::Range;
+
+use aster_block::{BlockDevice, id::Sid};
+use device_id::DeviceId;
+
+use self::{linear::LinearTarget, striped::StripedTarget};
+
+pub mod linear;
+pub mod striped;
+
+#[derive(Debug)]
+pub enum DmTarget {
+    Linear(LinearTarget),
+    Striped(StripedTarget),
+}
+
+impl DmTarget {
+    pub fn logical_range(&self) -> &Range<Sid> {
+        match self {
+            Self::Linear(target) => target.logical_range(),
+            Self::Striped(target) => target.logical_range(),
+        }
+    }
+
+    pub fn length(&self) -> u64 {
+        match self {
+            Self::Linear(target) => target.length(),
+            Self::Striped(target) => target.length(),
+        }
+    }
+
+    pub fn backing_start(&self) -> Option<Sid> {
+        match self {
+            Self::Linear(target) => Some(target.backing_start()),
+            Self::Striped(_) => None,
+        }
+    }
+
+    pub fn backing_id(&self) -> Option<DeviceId> {
+        match self {
+            Self::Linear(target) => Some(target.backing_id()),
+            Self::Striped(_) => None,
+        }
+    }
+
+    pub fn for_each_backing_id(&self, mut f: impl FnMut(DeviceId)) {
+        match self {
+            Self::Linear(target) => f(target.backing_id()),
+            Self::Striped(target) => target.for_each_backing_id(f),
+        }
+    }
+
+    pub fn backing(&self) -> Option<&dyn BlockDevice> {
+        match self {
+            Self::Linear(target) => Some(target.backing()),
+            Self::Striped(_) => None,
+        }
+    }
+
+    pub fn for_each_backing<'a>(&'a self, mut f: impl FnMut(&'a dyn BlockDevice)) {
+        match self {
+            Self::Linear(target) => f(target.backing()),
+            Self::Striped(target) => target.for_each_backing(f),
+        }
+    }
+
+    pub fn map_sector(&self, logical: Sid) -> Option<Sid> {
+        match self {
+            Self::Linear(target) => target.map_sector(logical),
+            Self::Striped(_) => None,
+        }
+    }
+
+    pub fn is_linear(&self) -> bool {
+        matches!(self, Self::Linear(_))
+    }
+}
