diff --git a/kernel/core/comps/device-mapper/Cargo.toml b/kernel/core/comps/device-mapper/Cargo.toml
new file mode 100644
index 000000000..72c83fc3e
--- /dev/null
+++ b/kernel/core/comps/device-mapper/Cargo.toml
@@ -0,0 +1,14 @@
+[package]
+name = "aster-device-mapper"
+version = "0.1.0"
+edition.workspace = true
+
+[dependencies]
+aster-block.workspace = true
+device-id.workspace = true
+id-alloc.workspace = true
+io-util.workspace = true
+ostd.workspace = true
+
+[lints]
+workspace = true
