diff --git a/kernel/core/comps/block/src/partition.rs b/kernel/core/comps/block/src/partition.rs
index 62d2d2e28..fdb0a5eaa 100644
--- a/kernel/core/comps/block/src/partition.rs
+++ b/kernel/core/comps/block/src/partition.rs
@@ -251,7 +251,7 @@ pub struct PartitionNode {
 
 impl BlockDevice for PartitionNode {
     fn enqueue(&self, mut bio: SubmittedBio) -> Result<(), BioEnqueueError> {
-        bio.set_sid_offset(self.info.start_sector());
+        bio.add_sid_offset(self.info.start_sector())?;
         self.device.enqueue(bio)
     }
 
@@ -261,8 +261,8 @@ impl BlockDevice for PartitionNode {
         metadata
     }
 
-    fn name(&self) -> &str {
-        &self.name
+    fn name(&self) -> String {
+        self.name.clone()
     }
 
     fn id(&self) -> DeviceId {
