diff --git a/kernel/comps/nvme/src/lib.rs b/kernel/comps/nvme/src/lib.rs
index eee435bfb..261b64344 100644
--- a/kernel/comps/nvme/src/lib.rs
+++ b/kernel/comps/nvme/src/lib.rs
@@ -39,7 +39,8 @@ static NVME_BLOCK_MAJOR_ID: Once<MajorIdOwner> = Once::new();
 
 #[init_component]
 fn nvme_init() -> Result<(), ComponentInitError> {
-    let major = aster_block::allocate_major().map_err(|_| ComponentInitError::Unknown)?;
+    let major =
+        aster_block::allocate_major_with_name("nvme").map_err(|_| ComponentInitError::Unknown)?;
     NVME_BLOCK_MAJOR_ID.call_once(|| major);
 
     transport::init();
