diff --git a/kernel/comps/virtio/src/lib.rs b/kernel/comps/virtio/src/lib.rs
index c8f42c108..0356cd69c 100644
--- a/kernel/comps/virtio/src/lib.rs
+++ b/kernel/comps/virtio/src/lib.rs
@@ -42,7 +42,7 @@ static VIRTIO_BLOCK_MAJOR_ID: Once<MajorIdOwner> = Once::new();
 
 #[init_component]
 fn virtio_component_init() -> Result<(), ComponentInitError> {
-    VIRTIO_BLOCK_MAJOR_ID.call_once(|| aster_block::allocate_major().unwrap());
+    VIRTIO_BLOCK_MAJOR_ID.call_once(|| aster_block::allocate_major_with_name("virtblk").unwrap());
 
     // Find all devices and register them to the corresponding crate
     transport::init();
