diff --git a/kernel/core/comps/nvme/src/device/block_device.rs b/kernel/core/comps/nvme/src/device/block_device.rs
index cf652f580..1bdd09d9b 100644
--- a/kernel/core/comps/nvme/src/device/block_device.rs
+++ b/kernel/core/comps/nvme/src/device/block_device.rs
@@ -84,8 +84,8 @@ impl aster_block::BlockDevice for NvmeBlockDevice {
         }
     }
 
-    fn name(&self) -> &str {
-        &self.name
+    fn name(&self) -> String {
+        self.name.clone()
     }
 
     fn id(&self) -> DeviceId {
