diff --git a/kernel/core/src/device/registry/char.rs b/kernel/core/src/device/registry/char.rs
index 4098813a4..c3502d8d8 100644
--- a/kernel/core/src/device/registry/char.rs
+++ b/kernel/core/src/device/registry/char.rs
@@ -15,6 +15,7 @@ use crate::{
 static DEVICE_REGISTRY: Mutex<BTreeMap<u32, Arc<dyn Device>>> = Mutex::new(BTreeMap::new());
 
 /// Registers a new char device.
+/// 将某个对象(拥有DeviceId) 登记到全局字符设备表中
 pub(crate) fn register(device: Arc<dyn Device>) -> Result<()> {
     let mut registry = DEVICE_REGISTRY.lock();
     let id = device.id().to_raw();
@@ -110,6 +111,7 @@ impl Drop for MajorIdOwner {
     }
 }
 
+//第一个进程 把之前注册好的设备对象 映射成/dev/xxx路径 根据devtmpfs_meta
 pub(super) fn init_in_first_process(path_resolver: &PathResolver) -> Result<()> {
     for device in collect_all() {
         if let Some(devtmpfs_meta) = device.devtmpfs_meta() {
