diff --git a/kernel/core/src/device/misc/mod.rs b/kernel/core/src/device/misc/mod.rs
index aa0e0e629..93be38cac 100644
--- a/kernel/core/src/device/misc/mod.rs
+++ b/kernel/core/src/device/misc/mod.rs
@@ -9,15 +9,18 @@ use spin::Once;
 
 use super::registry::char::{MajorIdOwner, acquire_major};
 
+mod device_mapper;
 mod hwrng;
 #[cfg(all(target_arch = "x86_64", feature = "cvm_guest"))]
 pub(crate) mod tdxguest;
 
 static MISC_MAJOR: Once<MajorIdOwner> = Once::new();
 
+// Linux misc devices use character major 10.
 pub(super) fn init_in_first_kthread() {
     MISC_MAJOR.call_once(|| acquire_major(MajorId::new(10)).unwrap());
 
+    device_mapper::init_in_first_kthread();
     hwrng::init_in_first_kthread();
 
     #[cfg(target_arch = "x86_64")]
