diff --git a/kernel/core/src/device/registry/char.rs b/kernel/core/src/device/registry/char.rs
index 4098813a4..905fa3a27 100644
--- a/kernel/core/src/device/registry/char.rs
+++ b/kernel/core/src/device/registry/char.rs
@@ -59,7 +59,8 @@ static MAJORS: Mutex<BTreeSet<u16>> = Mutex::new(BTreeSet::new());
 /// Acquires a major ID.
 ///
 /// The returned `MajorIdOwner` object represents the ownership to the major ID.
-/// Until the object is dropped, this major ID cannot be acquired via `acquire_major` or `allocate_major` again.
+/// Until the object is dropped, this major ID cannot be acquired via
+/// `acquire_major` or `allocate_major` again.
 pub(crate) fn acquire_major(major: MajorId) -> Result<MajorIdOwner> {
     if major.get() > MAX_MAJOR {
         return_errno_with_message!(Errno::EINVAL, "the major ID is invalid");
@@ -75,7 +76,8 @@ pub(crate) fn acquire_major(major: MajorId) -> Result<MajorIdOwner> {
 /// Allocates a major ID.
 ///
 /// The returned `MajorIdOwner` object represents the ownership to the major ID.
-/// Until the object is dropped, this major ID cannot be acquired via `acquire_major` or `allocate_major` again.
+/// Until the object is dropped, this major ID cannot be acquired via
+/// `acquire_major` or `allocate_major` again.
 #[expect(dead_code)]
 pub(crate) fn allocate_major() -> Result<MajorIdOwner> {
     let mut majors = MAJORS.lock();
@@ -110,6 +112,8 @@ impl Drop for MajorIdOwner {
     }
 }
 
+// The first userspace process materializes pre-registered devices under `/dev`
+// according to their devtmpfs metadata.
 pub(super) fn init_in_first_process(path_resolver: &PathResolver) -> Result<()> {
     for device in collect_all() {
         if let Some(devtmpfs_meta) = device.devtmpfs_meta() {
