diff --git a/test/initramfs/src/regression/fs/run_test.sh b/test/initramfs/src/regression/fs/run_test.sh
index 0507421ed..0e7b76cbc 100755
--- a/test/initramfs/src/regression/fs/run_test.sh
+++ b/test/initramfs/src/regression/fs/run_test.sh
@@ -131,6 +131,7 @@ echo "All mount bind file test passed."
 ./overlayfs/readdir_small_buffer
 
 ./procfs/dentry_cache
+./procfs/devices
 ./procfs/fd
 ./procfs/getdents
 ./procfs/mountstats
