diff --git a/test/initramfs/src/regression/fs/run_test.sh b/test/initramfs/src/regression/fs/run_test.sh
index 0a11a1d08..2d900ad05 100755
--- a/test/initramfs/src/regression/fs/run_test.sh
+++ b/test/initramfs/src/regression/fs/run_test.sh
@@ -132,6 +132,7 @@ echo "All mount bind file test passed."
 ./overlayfs/readdir_small_buffer
 
 ./procfs/dentry_cache
+./procfs/devices
 ./procfs/fd
 ./procfs/getdents
 ./procfs/mountstats
